File: 1. Data_Preprocessing/data_processing_template_other_steps.R
# Data Processing

# Importing dataset

dataset = read.csv('Data.csv')

# replacing missing data in Age column
dataset$Age = ifelse(is.na(dataset$Age),
                     ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)),
                     dataset$Age)
# replacing missing data in Salary column
dataset$Salary = ifelse(is.na(dataset$Salary),
                        ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
                        dataset$Salary)

# Encoding categorial data
dataset$Country = factor(dataset$Country, 
                         levels = c('France', 'Spain', 'Germany'), 
                         labels = c(1, 2, 3))

dataset$Purchased = factor(dataset$Purchased, 
                           levels = c('No', 'Yes'), 
                           labels = c(0, 1))

#install caTools libraries
#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Purchased, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# feature scaling
# specifying the second and third column to scale because first column(country) contains only country names(encoded to 1,2,3)
training_set[, 2:3] = scale(training_set[,2:3])
test_set[, 2:3] = scale(test_set[, 2:3])

File: 1. Data_Preprocessing/data_processing_template_other_steps.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Fri Jan 18 17:59:07 2019

@author: kodtodya
"""
# Data processing

# Importing libraries 

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing datasets

dataset = pd.read_csv('Data.csv')
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 3].values

# Taking care of missing data
from sklearn.preprocessing import Imputer

# Imputer clss transforms the missing vales based on strategy and axis(rows/columns)
imputer = Imputer(missing_values = 'NaN', strategy = 'mean', axis = 0)

# we are putting x into imputer because x contains the training data
imputer = imputer.fit(x[:, 1:3])
# Transforming the missing data with column mean and updating it
x[:, 1:3] = imputer.transform(x[:, 1:3])

# Encoding categorial data
from sklearn.preprocessing import LabelEncoder, OneHotEncoder

labelencoder_x = LabelEncoder()
x[:, 0] = labelencoder_x.fit_transform(x[:, 0])

onehotencode = OneHotEncoder(categorical_features=[0])
x = onehotencode.fit_transform(x).toarray()

labelencoder_y = LabelEncoder()
y = labelencoder_y.fit_transform(y)

# splitting dataset into Training set and test set
from sklearn.model_selection import train_test_split
x_train, x_test,y_train,y_test = train_test_split(x, y, test_size = 0.2, random_state = 0)

# feature scaling
from sklearn.preprocessing import StandardScaler
  
sc_x = StandardScaler()
x_train = sc_x.fit_transform(x_train)
x_ttest = sc_x.transform(x_test)
 